rs/tests/networking/state_sync_performance.rs

```diff
@@ -0,0 +1,214 @@
+};
+use slog::info;
+use std::time::Duration;
+
+const INITIAL_NODES: usize = 1;
+const TOTAL_NODES: usize = 13;
+
+const BANDWIDTH_MBITS: u32 = 300; // artificial cap on bandwidth
+const LATENCY: Duration = Duration::from_millis(150); // artificial added latency
+
+const SIZE_LEVEL: usize = 1;
+const NUM_CANISTERS: usize = 4;
+
+pub const SUCCESSFUL_STATE_SYNC_DURATION_SECONDS_SUM: &str =
+    "state_sync_duration_seconds_sum{status=\"ok\"}";
+pub const SUCCESSFUL_STATE_SYNC_DURATION_SECONDS_COUNT: &str =
+    "state_sync_duration_seconds_count{status=\"ok\"}";
+
+fn setup(env: TestEnv) {
+    PrometheusVm::default()
+        .start(&env)
+        .expect("failed to start prometheus VM");
+
+    InternetComputer::new()
+        .with_default_vm_resources(VmResources {
+            boot_image_minimal_size_gibibytes: Some(ImageSizeGiB::new(
+                30 + 2 * NUM_CANISTERS as u64 * SIZE_LEVEL as u64,
+            )),
+            ..VmResources::default()
+        })
+        .with_required_host_features(vec![HostFeature::IoPerformance])
+        .add_subnet(
+            Subnet::new(SubnetType::System)
+                .with_dkg_interval_length(Height::from(99))
+                .with_unit_delay(Duration::from_millis(200))
+                .with_initial_notary_delay(Duration::from_millis(200))
+                .add_nodes(INITIAL_NODES),
+        )
+        .with_unassigned_nodes(TOTAL_NODES - INITIAL_NODES)
+        .setup_and_start(&env)
+        .expect("failed to setup IC under test");
+
+    let topology = env.topology_snapshot();
+
+    for subnet in topology.subnets() {
+        for node in subnet.nodes() {
+            node.await_status_is_healthy()
+                .expect("Nodes failed to come up healthy")
+        }
+    }
+
+    for node in topology.unassigned_nodes() {
+        node.await_can_login_as_admin_via_ssh()
+            .expect("Timeout while waiting for all unassigned nodes to be healthy");
+    }
+
+    topology.root_subnet().apply_network_settings(
+        FixedNetworkSimulation::new()
+            .with_latency(LATENCY)
+            .with_bandwidth(BANDWIDTH_MBITS),
+    );
+    env.sync_with_prometheus();
+
+    let nns_node = topology.root_subnet().nodes().next().unwrap();
+    NnsInstallationBuilder::new()
+        .install(&nns_node, &env)
+        .expect("Failed to install NNS canisters");
+}
+
+fn test(env: TestEnv) {
+    let logger = env.logger();
+    let topology = env.topology_snapshot();
+    let nns_node = topology.root_subnet().nodes().next().unwrap();
+    let nns = runtime_from_url(nns_node.get_public_url(), nns_node.effective_canister_id());
+    let governance = Canister::new(&nns, GOVERNANCE_CANISTER_ID);
+
+    let agent_node = topology
+        .root_subnet()
+        .nodes()
+        .next()
+        .expect("Failed to get agent node");
+
+    block_on(async {
+        info!(
+            logger,
+            "Installing universal canister on a node {} ...",
```

Comment:
It looks below the test will install the statesync canister instead of the universal canister.